include/quickcpplib/algorithm/hash.hpp

```diff
@@ -48,21 +49,21 @@ namespace algorithm
     {
       size_t operator()(T v) const
       {
-#if defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__) || defined(__ia64__) || defined(_M_IA64) || defined(__ppc64__)
-        static constexpr size_t basis = 14695981039346656037ULL, prime = 1099511628211ULL;
-        static_assert(sizeof(size_t) == 8, "size_t is not 64 bit");
+#if QUICKCPPLIB_PLATFORM_NATIVE_BITLENGTH < 64
+        using working_type = uint32_t;
+        static constexpr working_type basis = 2166136261U, prime = 16777619U;
 #else
-        static constexpr size_t basis = 2166136261U, prime = 16777619U;
-        static_assert(sizeof(size_t) == 4, "size_t is not 32 bit");
+        using working_type = uint64_t;
+        static constexpr working_type basis = 14695981039346656037ULL, prime = 1099511628211ULL;
 #endif
         const unsigned char *_v = (const unsigned char *) &v;
-        size_t ret = basis;
+        working_type ret = basis;
         for(size_t n = 0; n < sizeof(T); n++)
         {
-          ret ^= (size_t) _v[n];
+          ret ^= (working_type) _v[n];
           ret *= prime;
         }
-        return ret;
+        return size_t(ret);
```

Comment:
This will trigger GCC's semi useful `[-Wuseless-cast]` if `typeof(size_t) == typeof(working_type)` holds. I haven't found a good way to deal with these occurences 🙄 (suppressing them in place is so much clutter).

Reply:
Yup looks like `size_t` and `uint64_t` have identical type definition. I'll fix this shortly.